pkg/proxy/proxy.go

```diff
@@ -142,10 +142,43 @@ func (pm *Manager) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, "no auth providers configured", http.StatusBadRequest)
 			return
 		}
-		sort.Slice(providers, func(i, j int) bool {
-			return providers[i] < providers[j]
+
+		sort.Slice(configuredProviders, func(i, j int) bool {
+			return configuredProviders[i] < configuredProviders[j]
 		})
-		provider = "default/" + providers[0]
+		provider = "default/" + configuredProviders[0]
+	} else {
+		namespace, name, _ := strings.Cut(provider, "/")
+		if namespace == "" {
+			http.Error(w, "invalid auth provider:"+provider, http.StatusBadRequest)
+			return
+		}
+
+		// Check if the provider is configured.
+		configuredProviders, err := pm.dispatcher.ListConfiguredAuthProviders(r.Context(), namespace)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("failed to list configured auth providers: %v", err), http.StatusInternalServerError)
+			return
+		}
+
+		if !slices.Contains(configuredProviders, name) {
+			// The requested auth provider isn't configured. Return an error, unless the user is signing out, in which case, just redirect.
+			if r.URL.Path == "/oauth2/sign_out" {
+				http.Redirect(w, r, rdParam, http.StatusFound)
```

Comment:
Should we clear the cookie in this case, too?

Reply:
Good call